src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -404,6 +406,23 @@ public async Task<RestoreResult> ExecuteAsync(CancellationToken token)
             }
         }
 
+        private bool HasValidPlatformVersions()
+        {
+            IEnumerable<NuGetFramework> badPlatforms = _request.Project.TargetFrameworks.Select(tfm => tfm.FrameworkName).Where(fw => !string.IsNullOrEmpty(fw.Platform) && (fw.PlatformVersion == FrameworkConstants.EmptyVersion));
+            if (badPlatforms.Any())
+            {
+                foreach (NuGetFramework fw in badPlatforms)
+                {
+                    _logger.Log(RestoreLogMessage.CreateError(NuGetLogCode.NU1012, string.Format(CultureInfo.CurrentCulture, Strings.Error_PlatformVersionNotPresent, fw.Framework, fw.Platform)));
```

Comment:
nit: In an ideal world, I'd recommend these get merged so that a user gets only 1 instead of N errors, but I simply don't see this scenario getting hit too often (actually I'd only expect it during development), so I would *not* bother too much :)